code/game/objects/items/devices/flashlight.dm

```diff
@@ -781,57 +781,69 @@
 	grind_results = list(/datum/reagent/phenol = 15, /datum/reagent/hydrogen = 10, /datum/reagent/oxygen = 5) //Meth-in-a-stick
 	sound_on = 'sound/effects/wounds/crack2.ogg' // the cracking sound isn't just for wounds silly
 	toggle_context = FALSE
-	/// How many seconds of fuel we have left
-	var/fuel = 0
 	/// How much max fuel we have
 	var/max_fuel = 0
+	/// How much oxygen gets added upon cracking the stick. Doesn't actually produce a reaction with the fluid but it does allow for bootleg chemical "grenades"
+	var/oxygen_added = 5
+	/// How much temperature gets added for every unit of fuel burned down
+	var/temp_per_fuel = 3
+	/// Type of reagent we add as fuel
+	var/fuel_type = /datum/reagent/luminescent_fluid
 	/// The timer id powering our burning
 	var/timer_id = TIMER_ID_NULL
 
-/obj/item/flashlight/glowstick/Initialize(mapload)
-	fuel = rand(20 MINUTES, 25 MINUTES)
-	max_fuel = fuel
+/obj/item/flashlight/glowstick/Initialize(mapload, fuel_override = null)
+	. = ..()
+	max_fuel = isnull(fuel_override) ? rand(20, 25) : fuel_override
+	create_reagents(max_fuel + oxygen_added, DRAWABLE | INJECTABLE)
+	reagents.add_reagent(fuel_type, max_fuel)
 	set_light_color(color)
-	return ..()
+	AddComponent(/datum/component/edible,\
+		food_flags = FOOD_NO_EXAMINE,\
+		volume = reagents.total_volume,\
+		bite_consumption = round(reagents.total_volume / (rand(20, 30) / 10)),\
```

Comment:
```suggestion
		bite_consumption = round(reagents.total_volume / (rand(2, 3))),\
```
This would lose a decimal point, but does that matter here?

Reply:
```suggestion
		bite_consumption = round(reagents.total_volume / (rand(20, 30) * 0.1)),\
```
If you want to keep the decimals.
I know this isn't a requirement anymore, but this more readable